tests/neural_compressor/run_glue_source.py

```diff
@@ -0,0 +1,617 @@
+        )
+    if training_args.do_train:
+        if "train" not in raw_datasets:
+            raise ValueError("--do_train requires a train dataset")
+        train_dataset = raw_datasets["train"]
+        if data_args.max_train_samples is not None:
+            max_train_samples = min(len(train_dataset), data_args.max_train_samples)
+            train_dataset = train_dataset.select(range(max_train_samples))
+
+    if training_args.do_eval:
+        if "validation" not in raw_datasets and "validation_matched" not in raw_datasets:
+            raise ValueError("--do_eval requires a validation dataset")
+        eval_dataset = raw_datasets["validation_matched" if data_args.task_name == "mnli" else "validation"]
+        if data_args.max_eval_samples is not None:
+            max_eval_samples = min(len(eval_dataset), data_args.max_eval_samples)
+            eval_dataset = eval_dataset.select(range(max_eval_samples))
+
+    if training_args.do_predict or data_args.task_name is not None or data_args.test_file is not None:
+        if "test" not in raw_datasets and "test_matched" not in raw_datasets:
+            raise ValueError("--do_predict requires a test dataset")
+        predict_dataset = raw_datasets["test_matched" if data_args.task_name == "mnli" else "test"]
+        if data_args.max_predict_samples is not None:
+            max_predict_samples = min(len(predict_dataset), data_args.max_predict_samples)
+            predict_dataset = predict_dataset.select(range(max_predict_samples))
+
+    # Log a few random samples from the training set:
+    if training_args.do_train:
+        for index in random.sample(range(len(train_dataset)), 3):
+            logger.info(f"Sample {index} of the training set: {train_dataset[index]}.")
+
+    # Get the metric function
+    if data_args.task_name is not None:
+        metric = load_metric("glue", data_args.task_name)
+    else:
+        metric = load_metric("accuracy")
+
+    # You can define your custom compute_metrics function. It takes an `EvalPrediction` object (a namedtuple with a
+    # predictions and label_ids field) and has to return a dictionary string to float.
+    def compute_metrics(p: EvalPrediction):
+        preds = p.predictions[0] if isinstance(p.predictions, tuple) else p.predictions
+        preds = np.squeeze(preds) if is_regression else np.argmax(preds, axis=1)
+        if data_args.task_name is not None:
+            result = metric.compute(predictions=preds, references=p.label_ids)
+            if len(result) > 1:
+                result["combined_score"] = np.mean(list(result.values())).item()
+            return result
+        elif is_regression:
+            return {"mse": ((preds - p.label_ids) ** 2).mean().item()}
+        else:
+            return {"accuracy": (preds == p.label_ids).astype(np.float32).mean().item()}
+
+    # Data collator will default to DataCollatorWithPadding when the tokenizer is passed to Trainer, so we change it if
+    # we already did the padding.
```

Comment:
Why not using the latest examples scripts of transformers by cloning the repo, and verifying that the model is correctly saved and can be loaded/used for inference, this would make the tests more robust in my opinion

Reply:
Sure, I can make this change. If we do this it means that we do not need `run_glue_source.py` and `run_glue_target.py` anymore right? Because we'll clone the latest example scripts on the fly every time the UT runs.

Reply:
@echarlaix Neural Coder default enables `save_onnx_model=False` in the line `quantizer.quantize(quantization_config=quantization_config, save_directory="quantized_model", save_onnx_model=False)`, because Neural Coder is designed for on-the-fly optimizing the code and then executes the optimized code directly (instead of saving the model) and thus it does not save the model by nature. Would it be okay if we simply tests whether the optimized code executes finely?

Reply:
Thanks for iterating on it, it looks great ! The model will be saved after quantization in the `./quantized_model/` directory no ?

Reply:
Would love to have your opinion on adding a check to verify the model can be loaded (if that's feasible)

Reply:
Sure, let me check

Reply:
test of loading saved model has been added to the UT